lib/gitWorkflow.js

```diff
@@ -261,6 +333,19 @@ export class GitWorkflow {
     }
   }
 
+  /**
+   * Remove untracked files, to prevent tasks from seeing them
+   */
+  async hideUntrackedFiles(ctx) {
+    try {
+      for (const file of this.untrackedFiles) {
+        await rename(path.resolve(this.topLevelDir, file), this.getHiddenFilepath(file))
+      }
+    } catch (error) {
+      handleError(error, ctx, HideUntrackedFilesError)
+    }
+  }
+
```

Comment:
Let's forget this idea of renaming files into a hidden directory. Instead, we should use `git stash`.

Right now we use `git stash store && git stash create` which doesn't actually remove the files:

https://github.com/lint-staged/lint-staged/blob/dad5490949f31442e7580246a9fad5565257d6cb/lib/gitWorkflow.js#L227-L239

Instead, we can run `git stash` or `git stash --include-untracked` depending on these new options.

Reply:
Might actually be that the stash already includes all tracked files, so we would only need to run `git checkout` for them, and then later restore them from the stash.

Reply:
So we would need to add the `--include-untracked` git stash flag here if the new `--hide-untracked` option is used.